cdap-ui/app/features/hydrator/services/detail/stores/pipeline-detail-runs-store.js

```diff
@@ -113,7 +110,8 @@ angular.module(PKG.name + '.feature.hydrator')
     };
     this.emitChange = function() {
       this.changeListeners.forEach(function(callback) {
-        callback(this.state);
+        var stateCopy = angular.copy(this.state);
```

Comment:
Very minor but we can do ```angular.copy``` outside the loop so that we could avoid doing it #of listener times :smile: No biggy. Just noticed it.